### PR TITLE
Force users to select an autocomplete value

### PR DESCRIPTION
**Pull request by Manoj-nathwani**
See [Jira ticket](https://fjelltopp.atlassian.net/browse/ADX-394?atlOrigin=eyJpIjoiNDhjY2I0ODUxYmQxNGQxNjkwZThjMjVjMmJhOGY3ZjkiLCJwIjoiaiJ9)

### CODE REVIEW

File: ckanext/scheming/fanstatic/autocomplete-without-creating-new-options.js [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+ *            from the ajax returned results
+ * label    - A string of the label you want to appear within the dropdown for
+ *            returned results
+ *
+ * Examples
+ *
+ *   // <input name="tags" data-module="autocomplete" data-module-source="http://" />
+ *
+ */
+this.ckan.module('autocomplete-without-creating-new-options', function (jQuery) {
```
Comment (Manoj-nathwani) [inline ckanext/scheming/fanstatic/autocomplete-without-creating-new-options.js:18]: I've changed this line

File: ckanext/scheming/fanstatic/autocomplete-without-creating-new-options.js [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+      }
+      else {
+        if (/MSIE (\d+\.\d+);/.test(navigator.userAgent)) {
+            var ieversion=new Number(RegExp.$1);
+            if (ieversion<=7) {return}
+         }
+      }
+
+      // disable creating new options
+      settings.createSearchChoice = false;
```
Comment (Manoj-nathwani) [inline ckanext/scheming/fanstatic/autocomplete-without-creating-new-options.js:75]: And this line

**Review by jonathansberry — COMMENTED** [review]
I've been discussing this with Manoj, so approve of the route he has taken he. 

Just want to acknowledge that this goes hand in hand with his other PR: https://github.com/fjelltopp/unaids_data_specifications/pull/6

In the organisation membership manage page, there is an auto complete to select usernames for users to be added to the organisation. We should add this module there as well. It will probably be a PR in a different repo.

@tomeksabala The big question on my head is where should this js module reside and be registered?  Should it be in scheming or in unaids extension?  What do you think?  Moving it to UNAIDS would be a 5 min job.

**Review by tomeksabala — COMMENTED** [review]
Is there no easy way to just decorate the original `this.ckan.module('autocomplete')`?
Or at least overwrite only the relevant function `setupAutoComplete` and reuse the remaining ones.

**Comment by tomeksabala** [comment]
@jonathansberry I think it'd be better placed in ckanext-unaids.

**Comment by jonathansberry** [comment]
> Is there no easy way to just decorate the original `this.ckan.module('autocomplete')`?
> Or at least overwrite only the relevant function `setupAutoComplete` and reuse the remaining ones.

I've never seen this done anywhere else.  I think we have a solution that is fairly uncoupled from ckan core.  If CKAN core updates the auto-complete module our module won't be broken by that.  But we can update our module at our lesuire as necessary.  I think this is the right route to go for now.

**Comment by jonathansberry** [comment]
> @jonathansberry I think it'd be better placed in ckanext-unaids.

Ok @Manoj-nathwani let's move this to ckanext-unaids then, and also catch the add member form change as well.

**Comment by Manoj-nathwani** [comment]
Moved this logic to https://github.com/fjelltopp/ckanext-unaids/pull/67

**Pull request closed without merging**